src/Network/Transport/TCP.hs

```diff
@@ -1450,7 +1452,7 @@ schedule theirEndPoint act = do
 -- 'runScheduledAction' or by another).
 runScheduledAction :: EndPointPair -> Action a -> IO a
 runScheduledAction (ourEndPoint, theirEndPoint) mvar = do
-    join $ readChan (remoteScheduled theirEndPoint)
+    join $ uninterruptibleMask_ $ readChan (remoteScheduled theirEndPoint)
```

Comment:
> `mask_ $ readChan ...` can be interrupted when the channel is not empty according to the implementation of `readChan`.

I don't see it. In fact empirically, the following program doesn't terminate:

```Haskell
main = do
    chan <- newChan
    tid <- forkIO $ mask_ $ forever $ writeChan chan () >> readChan chan
    putStrLn "start"
    killThread tid
    putStrLn "stop"
```

Not that I'm too concerned about that - as I said in a previous comment, I'd prefer setting exception handlers or using resourcet rather than masking whole sections of code. Unless it can be demonstrated that doing so has a performance impact.